### PR TITLE
Preload group users

### PR DESCRIPTION
**Pull request by JosteinLindhom**

### CODE REVIEW

**Comment by meling** [comment]
This looks like a nice cleanup. Could you create an issue that briefly explains the cleanup being done and link to this PR. Also, would be nice to have some tests for these features.

File: database/gormdb_group.go [hunk 1]
```diff
@@ -136,31 +136,27 @@ func (db *GormDB) DeleteGroup(groupID uint64) error {
...
-		userIds = append(userIds, enrollment.UserID)
-		u, err := db.GetUser(enrollment.UserID)
-		if err != nil {
-			return nil, err
-		}
-		enrollment.User = u
-	}
-	if len(userIds) == 0 {
+	if len(userIDs) == 0 {
 		return nil, errors.New("failed to get next student reviewer: no users in group")
```
Comment (meling) [inline database/gormdb_group.go:158]: This error message seems a bit too specific. Should be more generic.

File: database/gormdb_group.go [hunk 2]
```diff
@@ -136,31 +136,27 @@ func (db *GormDB) DeleteGroup(groupID uint64) error {
...
+	if err := db.conn.
+		Preload("Enrollments").
+		Preload("Enrollments.UsedSlipDays").
+		Preload("Enrollments.User").
+		Preload("Users", "id IN ?", userIDs).
+		First(&group, groupID).Error; err != nil {
 		if err == gorm.ErrRecordNotFound {
 			return nil, err
 		}
 		return nil, fmt.Errorf("error fetching group record for group with ID %d: %w", groupID, err)
```
Comment (meling) [inline database/gormdb_group.go:155]: Error message could also be shortened

File: database/gormdb_group_test.go [hunk 3]
```diff
@@ -202,12 +202,6 @@ func TestGormDBCreateAndGetGroup(t *testing.T) {
 				}
 			}
 			group.Enrollments = enrollments
-			for _, usr := range have.Users {
-				usr.Enrollments = nil
-			}
-			for _, e := range have.Enrollments {
-				e.User.Enrollments = nil
-			}
```
Comment (meling) [inline database/gormdb_group_test.go:210]: Let me know if tests are sufficient already.

**Review by meling — CHANGES_REQUESTED** [review]
Some changes while we are touching this is to fix the error messages.

File: database/gormdb_group_test.go (hunk 3 above)
Comment (JosteinLindhom) [reply database/gormdb_group_test.go:210]: I believe `TestGormDBCreateAndGetGroup` and `TestGetGroupsByCourse` are sufficient already.

**Review by meling — APPROVED** [review]

**Pull request merged**